src/core/room.ts

```diff
@@ -257,17 +389,17 @@ export class DefaultRoom implements Room {
   /**
    * @inheritdoc Room
    */
-  async attach(): Promise<void> {
+  async attach() {
     this._logger.trace('Room.attach();');
-    return this._lifecycleManager.attach();
+    return this._asyncOpsAfter.then(() => this._lifecycleManager?.attach());
```

Comment:
Do we want to ensure the `_lifecycleManager` is not undefined here? We could then return a better defined error, what do you think?

Reply:
Ah I see, I suppose it's not necessary then if `_lifecycleManager` is guaranteed to be there after `_asyncOpsAfter` resolves :)